### ISSUE TITLE
`spaced-comment` should allow line ending after markers

### ISSUE THREAD

**Issue by tiansh** [eslint/eslint#12036]
**What rule do you want to change?**
[spaced-comment](https://eslint.org/docs/rules/spaced-comment)

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
default behavior

**Please provide some example code that this change will affect:**

```js
/* eslint spaced-comment: ["error", "always", { "line": { "markers": ["/", "#region", "#endregion"] } }] */
/* eslint no-trailing-spaces: ["error"] */

//#region blah
const answer = 42;
//#endregion

console.log(answer);
```

**What does the rule currently do for this code?**

> 6:1  error  Expected space or tab after '//#endregion' in comment  spaced-comment

using `--fix` cannot fix it (since there are no possible auto fix)

**What will the rule do after it's changed?**

pass without tip an error

**Are you willing to submit a pull request to implement this change?**

Maybe no

----

`spaced-comment` does allow line ending after `//` without using markers. And this behavior should be applied to markers too imo.

**Comment by platinumazure** [comment]
This feels like a bug to me. I think it shouldn't matter if a comment has markers after the content or not.

I'm going to relabel this as a bug, but team members can feel free to correct that if needed.

**Comment by platinumazure** [comment]
Reproduction [here](eyJ0ZXh0IjoiLyogZXNsaW50IHNwYWNlZC1jb21tZW50OiBbXCJlcnJvclwiLCBcImFsd2F5c1wiLCB7IFwibGluZVwiOiB7IFwibWFya2Vyc1wiOiBbXCIvXCIsIFwiI3JlZ2lvblwiLCBcIiNlbmRyZWdpb25cIl0gfSB9XSAqL1xuLyogZXNsaW50IG5vLXRyYWlsaW5nLXNwYWNlczogW1wiZXJyb3JcIl0gKi9cblxuLy8jcmVnaW9uIGJsYWhcbmNvbnN0IGFuc3dlciA9IDQyO1xuLy8jZW5kcmVnaW9uXG5cbmNvbnNvbGUubG9nKGFuc3dlcik7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19), so marking as accepted.

**Comment by g-plane** [comment]
@platinumazure Repro link is broken.

**Comment by platinumazure** [comment]
@g-plane Sorry, bad paste. It's fixed now.

**Comment by mdjermanovic** [comment]
I'm working on this, but have a question. Should the rule also allow these:

```js
/* eslint spaced-comment: ["error", "always", { "block": { "markers": ["foo"] } }] */

/*foo*/
```

```js
/* eslint spaced-comment: ["error", "always", { "block": { "markers": ["foo"], balanced: true } }] */

/*foo*/
```

I.e. ignore a comment that contains only a marker like it already ignores an empty `/**/`

The issue with `//#endregion` is that the rule requires a `\s` after the marker, but there is nothing. Linebreak is not included in comment's `value`. 

`\\` immediately followed by the line ending is allowed simply because it's an empty comment.

**Comment by DmytroLapshyn** [comment]
If anyone cares for a workaround, this worked for me:

```javascript
"markers": ["#region"],
"exceptions": ["#endregion"]
```
